javaslang-benchmark/src/test/java/javaslang/JmhRunner.java

```diff
@@ -71,9 +72,28 @@ private static void runAndReport(Class<?> benchmarkClass, int warmupIterations,
         }
     }
 
-    public static <T> void assertEquals(T a, T b) {
-        if (!Objects.equals(a, b)) {
-            throw new IllegalStateException(a + " != " + b);
+    public static <T> void require(T value, Predicate<T> predicate) {
+        require(() -> predicate.test(value));
+    }
+
+    public static <T1, T2> void require(T1 value1, Predicate<T1> predicate1, T2 value2, Predicate<T2> predicate2) {
+        require(() -> predicate1.test(value1),
+                () -> predicate2.test(value2));
+    }
+
+    public static <T1, T2, T3> void require(T1 value1, Predicate<T1> predicate1, T2 value2, Predicate<T2> predicate2, T3 value3, Predicate<T3> predicate3) {
+        require(() -> predicate1.test(value1),
+                () -> predicate2.test(value2),
+                () -> predicate3.test(value3));
+    }
+
+    public static void require(BooleanSupplier... suppliers) {
```

Comment:
@danieldietrich, @eduardmanas, you're both right :)

for
```
final String s = "hello";
require(() -> !s.isEmpty());
return s;
```
the difference is already measurable (though still quite low), i.e.:
```
Target           Operation   Impl            Params  Count            Score  ±     Error    Unit      a      b
VectorBenchmark  DeadCode    a                          12  277,539,221.199  ±     0.54%   ops/s        0.90x
VectorBenchmark  DeadCode    b                          12  308,182,430.575  ±     0.90%   ops/s 1.11x 
```

However :) ...
Since we've already disabled assertions for normal run, it would actually be a **FEATURE** to run the quick debug WITH ALL ASSERTIONS enabled :D.
I didn't realize this until now, thanks for your help. Will change all `require` calls to assertions (tested it with JMH, they're indeed completely eliminated)